java/flight/flight-core/src/main/java/org/apache/arrow/flight/example/integration/AuthBasicProtoScenario.java

```diff
@@ -58,40 +58,37 @@ public void doAction(CallContext context, Action action, StreamListener<Result>
   public void buildServer(FlightServer.Builder builder) {
     builder.authHandler(new BasicServerAuthHandler(new BasicServerAuthHandler.BasicAuthValidator() {
       @Override
-      public byte[] getToken(String username, String password) throws Exception {
+      public Optional<String> validateCredentials(String username, String password) throws Exception {
+        if (Strings.isNullOrEmpty(username)) {
+          throw CallStatus.UNAUTHORIZED.withDescription("Credentials not supplied").toRuntimeException();
+        }
+
         if (!USERNAME.equals(username) || !PASSWORD.equals(password)) {
-          throw CallStatus.UNAUTHENTICATED.withDescription("Username or password is invalid.").toRuntimeException();
+          throw CallStatus.UNAUTHORIZED.withDescription("Username or password is invalid.").toRuntimeException();
         }
-        return ("valid:" + username).getBytes(StandardCharsets.UTF_8);
+        return Optional.of("valid:" + username);
       }
 
       @Override
-      public Optional<String> isValid(byte[] token) {
-        if (token != null) {
-          final String credential = new String(token, StandardCharsets.UTF_8);
-          if (credential.startsWith("valid:")) {
-            return Optional.of(credential.substring(6));
-          }
+      public Optional<String> isValid(String token) {
+        if (token.startsWith("valid:")) {
+          return Optional.of(token.substring(6));
         }
         return Optional.empty();
       }
     }));
   }
 
   @Override
-  public void client(BufferAllocator allocator, Location location, FlightClient client) {
-    final FlightRuntimeException e = IntegrationAssertions.assertThrows(FlightRuntimeException.class, () -> {
-      client.listActions().forEach(act -> {
-      });
-    });
-    if (!FlightStatusCode.UNAUTHENTICATED.equals(e.status().code())) {
-      throw new AssertionError("Expected UNAUTHENTICATED but found " + e.status().code(), e);
-    }
-
-    client.authenticate(new BasicClientAuthHandler(USERNAME, PASSWORD));
-    final Result result = client.doAction(new Action("")).next();
-    if (!USERNAME.equals(new String(result.getBody(), StandardCharsets.UTF_8))) {
-      throw new AssertionError("Expected " + USERNAME + " but got " + Arrays.toString(result.getBody()));
+  public void client(BufferAllocator allocator, Location location, FlightClient.Builder clientBuilder) {
```

Comment:
We should still test what happens if the client tries to make an unauthenticated request.